RecoTracker/LSTCore/src/alpaka/Quadruplet.h

```diff
@@ -0,0 +1,1065 @@
+                                                            float regressionRadius,
+                                                            float nonAnchorRegressionRadius) {
+    quadruplets.tripletIndices()[quadrupletIndex][0] = innerTripletIndex;
+    quadruplets.tripletIndices()[quadrupletIndex][1] = outerTripletIndex;
+
+    quadruplets.lowerModuleIndices()[quadrupletIndex][0] = lowerModule1;
+    quadruplets.lowerModuleIndices()[quadrupletIndex][1] = lowerModule2;
+    quadruplets.lowerModuleIndices()[quadrupletIndex][2] = lowerModule3;
+    quadruplets.lowerModuleIndices()[quadrupletIndex][3] = lowerModule4;
+    quadruplets.innerRadius()[quadrupletIndex] = __F2H(innerRadius);
+    quadruplets.outerRadius()[quadrupletIndex] = __F2H(outerRadius);
+    quadruplets.pt()[quadrupletIndex] = __F2H(pt);
+    quadruplets.eta()[quadrupletIndex] = __F2H(eta);
+    quadruplets.phi()[quadrupletIndex] = __F2H(phi);
+    quadruplets.score_rphisum()[quadrupletIndex] = __F2H(scores);
+    quadruplets.layer()[quadrupletIndex] = layer;
+    quadruplets.isDup()[quadrupletIndex] = 0;
+    quadruplets.logicalLayers()[quadrupletIndex][0] = triplets.logicalLayers()[innerTripletIndex][0];
+    quadruplets.logicalLayers()[quadrupletIndex][1] = triplets.logicalLayers()[innerTripletIndex][1];
+    quadruplets.logicalLayers()[quadrupletIndex][2] = triplets.logicalLayers()[innerTripletIndex][2];
+    quadruplets.logicalLayers()[quadrupletIndex][3] = triplets.logicalLayers()[outerTripletIndex][2];
+
+    quadruplets.hitIndices()[quadrupletIndex][0] = triplets.hitIndices()[innerTripletIndex][0];
+    quadruplets.hitIndices()[quadrupletIndex][1] = triplets.hitIndices()[innerTripletIndex][1];
+    quadruplets.hitIndices()[quadrupletIndex][2] = triplets.hitIndices()[innerTripletIndex][2];
+    quadruplets.hitIndices()[quadrupletIndex][3] = triplets.hitIndices()[innerTripletIndex][3];
+    quadruplets.hitIndices()[quadrupletIndex][4] = triplets.hitIndices()[innerTripletIndex][4];
+    quadruplets.hitIndices()[quadrupletIndex][5] = triplets.hitIndices()[innerTripletIndex][5];
+    quadruplets.hitIndices()[quadrupletIndex][6] = triplets.hitIndices()[outerTripletIndex][4];
+    quadruplets.hitIndices()[quadrupletIndex][7] = triplets.hitIndices()[outerTripletIndex][5];
+
+    quadruplets.rzChiSquared()[quadrupletIndex] = rzChiSquared;
+    quadruplets.dBeta()[quadrupletIndex] = dBeta;
+    quadruplets.promptScore()[quadrupletIndex] = promptScore;
+    quadruplets.displacedScore()[quadrupletIndex] = displacedScore;
+    quadruplets.fakeScore()[quadrupletIndex] = fakeScore;
+
+    quadruplets.regressionRadius()[quadrupletIndex] = regressionRadius;
+    quadruplets.nonAnchorRegressionRadius()[quadrupletIndex] = nonAnchorRegressionRadius;
+    quadruplets.regressionG()[quadrupletIndex] = regressionG;
+    quadruplets.regressionF()[quadrupletIndex] = regressionF;
+  };
+
+  template <typename TAcc>
+  ALPAKA_FN_ACC ALPAKA_FN_INLINE bool passT4RZConstraint(TAcc const& acc,
```

Comment:
check the updates (now in master) in T3 `passRZConstraint` and try to simplify the code here in line with that (charge and related quantities).
In a combination with the repeated code in `passT5RZConstraint`, perhaps there is a way to reuse the code as well